src/store/storage-controller.ts

```diff
@@ -111,19 +109,35 @@ export class StorageController {
     return messages;
   }
 
-  public static async delete(
-    messageStore: MessageStore,
-    dataStore: DataStore,
+  public async delete(
     tenant: string,
     message: BaseMessage
   ): Promise<void> {
     const messageCid = await Message.getCid(message);
 
     if (message.descriptor.dataCid !== undefined) {
-      await dataStore.delete(tenant, messageCid, message.descriptor.dataCid);
+      await this.dataStore.delete(tenant, messageCid, message.descriptor.dataCid);
     }
 
-    await messageStore.delete(tenant, messageCid);
+    await this.messageStore.delete(tenant, messageCid);
+  }
+
+  public get MessageStore(): MessageStore { return this.messageStore; }
+
+  public queryMessageStore(tenant: string, filter: Filter, options?: MessageStoreOptions): Promise<BaseMessage[]> {
+    return this.messageStore.query(tenant, filter, options);
+  }
+
+  public get(tenant: string, messageCid: string, dataCid: string): Promise<GetResult|undefined> {
+    return this.dataStore.get(tenant, messageCid, dataCid);
+  }
+
+  public deleteEventsByCid(tenant: string, deletedMessageCids: any): void {
+    if (this.eventLog) { this.eventLog.deleteEventsByCid(tenant, deletedMessageCids); }
```

Comment:
when does `this.eventLog` not exist?

also, should we `return` the result of this?

Reply:
I made the event log an optional argument of the StorageController constructor.

Reply:
is there a real usecase where it's not expected/required?

if the answer is "tests" IMO that's not a very good reason and we should just create it where needed (or not even use `StorageController` in the first place)

Reply:
Addressed in https://github.com/TBD54566975/dwn-sdk-js/pull/338/commits/8a39de45aaa23185e406373a304c4ee9e4d68730